e2e/scenario/vhd.go

```diff
@@ -1,116 +1,12 @@
-	// When a VHD build ID is specified, this catalog's entries will be overwritten respectively for each downloaded VHD publishing info.
-	BaseVHDCatalog = mustGetVHDCatalogFromEmbeddedJSON(embeddedBaseVHDCatalog)
-)
-
-func getVHDsFromBuild(ctx context.Context, tmpl *Template, scenarios []*Scenario) error {
-	if config.VHDBuildID == "" {
-		return nil
-	}
-
-	buildID, err := strconv.Atoi(config.VHDBuildID)
-	if err != nil {
-		return fmt.Errorf("unable to convert build ID %s to int: %w", config.VHDBuildID, err)
-	}
-
-	log.Printf("will use VHDs from specified build: %d", config.VHDBuildID)
-
-	downloader, err := artifact.NewDownloader(ctx)
-	if err != nil {
-		return fmt.Errorf("unable to construct new ADO artifact downloader: %w", err)
-	}
-
-	artifactNames := make(map[string]bool)
-	for _, scenario := range scenarios {
-		if scenario.VHDSelector == nil {
-			return fmt.Errorf("unable to download VHDs from build: scenario %q has an undefined VHDSelector", scenario.Name)
-		}
-		artifactName := scenario.VHDSelector().ArtifactName
-		if artifactName != "" && !artifactNames[artifactName] {
-			artifactNames[artifactName] = true
-			log.Printf("will download publishing info artifact for: %q", artifactName)
-		}
-	}
-
-	log.Printf("using artifact from build %d with name: %s", config.VHDBuildID, fmt.Sprint(artifactNames))
-
-	err = downloader.DownloadVHDBuildPublishingInfo(ctx, artifact.PublishingInfoDownloadOpts{
-		BuildID:       buildID,
-		TargetDir:     artifact.DefaultPublishingInfoDir,
-		ArtifactNames: artifactNames,
-	})
-	defer os.RemoveAll(artifact.DefaultPublishingInfoDir)
-	if err != nil {
-		return fmt.Errorf("unable to download VHD publishing info: %w", err)
-	}
-
-	if err = tmpl.VHDCatalog.addEntriesFromPublishingInfoDir(artifact.DefaultPublishingInfoDir); err != nil {
-		return fmt.Errorf("unable to load VHD selections from publishing info dir %s: %w", artifact.DefaultPublishingInfoDir, err)
-	}
-
-	return nil
-}
-
-// getVHDNameFromPublishingInfo will resolve the name of the VHD from the specified publishing info.
-// Resolved names will take the form of: 2204gen2containerd, v2gen2, azurelinuxv2gen2arm64, etc.
-func getVHDNameFromPublishingInfo(info artifact.VHDPublishingInfo) string {
-	vhdName := strings.ToLower(info.SKUName)
-	if info.OfferName == offerNameAzureLinux {
-		// explicitly prepend 'azurelinux' to azurelinux VHD names since their SKU
-		// names use the same naming convention as CBLMariner-based SKUs.
-		vhdName = fmt.Sprintf("azurelinux%s", vhdName)
-	}
-	return vhdName
-}
-
-func mustGetVHDCatalogFromEmbeddedJSON(rawJSON string) VHDCatalog {
-	if rawJSON == "" {
-		panic("default_vhd_catalog.json is empty")
-	}
-
-	catalog := VHDCatalog{}
-	if err := json.Unmarshal([]byte(rawJSON), &catalog); err != nil {
-		panic(err)
-	}
-
-	return catalog
-}
-
 type Manifest struct {
 	Containerd struct {
 		Edge string `json:"edge"`
```

Comment:
unrelated to these changes, but we should make a note that we'll need to update this based on the consolidation towards components.json

Reply:
yeah no worries, more-so just a note for myself and @Devinwong

Reply:
Noted!